### PR TITLE
Bar Graph - Update to support the allowXCallibration prop

### PR DESCRIPTION
**Pull request by artpark**
[UXPLATFORM-6814](https://jira2.cerner.com/browse/UXPLATFORM-6814)
===

### Summary
Allows a bar graph to be configured with the 'Allow Callibration' preference. This allows the user to see all the data points without it being cut off.

### Deployment Link
https://terra-graphs-deployed-pr-#.herokuapp.com/

### Testing
`npm run karma`
Screen Shot 2022-09-07 at 1 20 27 PM

Thank you for contributing to Terra.
@cerner/terra
@cerner/carbon

### CODE REVIEW

File: packages/carbon-graphs/CHANGELOG.md [hunk 1]
```diff
@@ -6,6 +6,7 @@
   * Added allowCalibration support for the x-axis on a line graph.
   * Finished implementation for `config.axis.x.allowCalibration` for a paired result graph with a numerical and timeseries x-axis.
   * Finished implementation for `config.axis.x.allowCalibration` for a scatter graph with a numerical and timeseries x-axis.
+  * Finished implementation for `config.axis.x.allowCalibration` for a bar graph with a numerical and timeseries x-axis.
```
Comment (sdadn) [inline packages/carbon-graphs/CHANGELOG.md:9]: I would update this to match: https://github.com/cerner/terra-graphs/blob/0db73f925acd20425bf93ab5f49c1b162126126d/packages/carbon-graphs/CHANGELOG.md?plain=1#L6-L8
```suggestion
  * Added support for `config.axis.x.allowCalibration` for a bar graph with a numerical and timeseries x-axis.
```

File: packages/carbon-graphs/src/js/controls/Bar/Bar.js [hunk 2]
```diff
@@ -34,22 +34,54 @@ import {
...
+    // for easier calculations
+    if (typeof i.x === 'string' || i.x instanceof Date) {
+      return utils.getEpocFromDateString(i.x);
+    }
+    return i.x;
+  });
+  return {
+    min: Math.min(...xAxisValuesList),
+    max: Math.max(...xAxisValuesList),
   };
```
Comment (sdadn) [inline packages/carbon-graphs/src/js/controls/Bar/Bar.js:85]: I would also update this to match https://github.com/cerner/terra-graphs/blob/0db73f925acd20425bf93ab5f49c1b162126126d/packages/carbon-graphs/src/js/controls/Bubble/helpers/helpers.js#L510-L530

**Comment by sdadn** [comment]
Was this tested for grouped bars? https://engineering.cerner.com/terra-ui/graphs/cerner-terra-graphs-docs/bar-graph/basic-bar-graphs/grouped-bars

File: packages/carbon-graphs/CHANGELOG.md (hunk 1 above)
Comment (artpark) [reply packages/carbon-graphs/CHANGELOG.md]: Fixed [149ec3e](https://github.com/cerner/terra-graphs/pull/293/commits/149ec3e9c1704f514ee29f51618f5840f4d650fb)

File: packages/carbon-graphs/src/js/controls/Bar/Bar.js (hunk 2 above)
Comment (artpark) [reply packages/carbon-graphs/src/js/controls/Bar/Bar.js]: Fixed [ab6c070](https://github.com/cerner/terra-graphs/pull/293/commits/ab6c0704180e22a19172a6ba2d7663360a47002a)

**Review by sycombs — APPROVED** [review]

**Review by sdadn — APPROVED** [review]

**Pull request merged**